stan/math/gpu/kernels/scalar_mul.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef STAN_MATH_GPU_KERNELS_SCALAR_MUL_HPP
+#define STAN_MATH_GPU_KERNELS_SCALAR_MUL_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/gpu/kernel_cl.hpp>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+// \cond
+const char *scalar_mul_kernel_code = STRINGIFY(
+    // \endcond
+    /**
+     * Multiplication of the matrix A with a scalar
+     *
+     * @param[in] A input matrix
+     * @param[in] B output matrix
+     * @param[in] scalar the value with which to multiply A
+     * @param[in] rows the number of rows in A
+     * @param[in] cols the number of columns in A
+     */
+    __kernel void scalar_mul(__global double *A, const __global double *B,
+                             const double scalar, const unsigned int rows,
+                             const unsigned int cols) {
+      int i = get_global_id(0);
+      int j = get_global_id(1);
+      if (i < rows && j < cols) {
```

Comment:
Same question here - can we not rely on the NDRange to do the bounds checking?

Reply:
@rok-cesnovar correct if if I'm wrong here, but not we can't. If we have a work group of size (32,32) and global of (34,34) that means 4 work groups would go off (the main chunk to handle index (0:31) then 3 others to handle indexes for (31:32). So we need this if statement to make sure those 3 work groups only work within the matrix

Reply:
You are right @SteveBronder. In these cases we dont even know what is the thread block size, as we leave it to the OpenCL implementation to handle the thread-block size. 

In order to avoid this we would have to set the thread-block size everywhere and pad the matrices the way we do it in the matrix multiplication. Since these kernels are pretty simple this checks do not present any real performance issues.